src/clients/QBittorrent.ts

```diff
@@ -215,76 +221,140 @@ export default class QBittorrent implements TorrentClient {
 	}
 
 	async isInfoHashInClient(infoHash: string): Promise<boolean> {
-		const responseText = await this.request(
-			"/torrents/properties",
-			`hash=${infoHash}`,
+		const torrent = await this.getTorrentInfo(infoHash);
+		return torrent.length > 0;
+	}
+
+	async addTorrent(formData: FormData): Promise<void> {
+		await this.request("/torrents/add", formData);
+	}
+
+	async recheckTorrent(infoHash: string): Promise<void> {
+		const torrent = await this.getTorrentInfo(infoHash);
+		if (torrent.length === 0) {
+			throw new Error("Torrent not found in client");
+		}
+		await this.request(
+			"/torrents/recheck",
+			`hashes=${torrent[0].hash}`,
 			X_WWW_FORM_URLENCODED,
 		);
```

Comment:
these helpers are great! they make substantially more calls to `this.getTorrentInfo` though. is there a way to minimize them?

Reply:
Yeah we could just pass torrent info as an arg and let the caller handle it.